presto-teradata-functions/src/main/java/com/facebook/presto/teradata/functions/TeradataDateFunctions.java

```diff
@@ -35,23 +39,30 @@
 import static com.facebook.presto.common.type.TimeZoneKey.getTimeZoneKeys;
 import static com.facebook.presto.spi.StandardErrorCode.GENERIC_INTERNAL_ERROR;
 import static com.facebook.presto.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
-import static com.facebook.presto.teradata.functions.dateformat.DateFormatParser.createDateTimeFormatter;
 import static com.google.common.base.Throwables.throwIfInstanceOf;
 import static io.airlift.slice.Slices.utf8Slice;
 import static java.nio.charset.StandardCharsets.UTF_8;
 import static java.util.concurrent.TimeUnit.MILLISECONDS;
 
 public final class TeradataDateFunctions
 {
+    private static final ThreadLocalCache<Slice, DateTimeFormatter> DATETIME_PARSER_CACHE =
+            new ThreadLocalCache<>(100, format -> DateFormatParser.createDateTimeParser(format.toStringUtf8()));
+
     private static final ThreadLocalCache<Slice, DateTimeFormatter> DATETIME_FORMATTER_CACHE =
-            new ThreadLocalCache<>(100, format -> createDateTimeFormatter(format.toStringUtf8()));
+            new ThreadLocalCache<>(100, format -> DateFormatParser.createDateTimeFormatter(format.toStringUtf8()));
 
-    private static final ISOChronology[] CHRONOLOGIES = new ISOChronology[MAX_TIME_ZONE_KEY + 1];
+    private static final ZoneId[] ZONE_IDS = new ZoneId[MAX_TIME_ZONE_KEY + 1];
 
     static {
         for (TimeZoneKey timeZoneKey : getTimeZoneKeys()) {
-            DateTimeZone dateTimeZone = DateTimeZone.forID(timeZoneKey.getId());
-            CHRONOLOGIES[timeZoneKey.getKey()] = ISOChronology.getInstance(dateTimeZone);
+            try {
+                ZONE_IDS[timeZoneKey.getKey()] = ZoneId.of(timeZoneKey.getId());
+            }
+            catch (ZoneRulesException ignored) {
```

Comment:
ZoneRulesException is a runtime exception. 
If we are catching it let us log the error so that we are not silently moving on.

Reply:
Log added in 2b89c16